src/planning.py

```diff
@@ -71,13 +66,17 @@ def sesame_plan(task: Task,
         # that initially has empty effects may later have a _NOT_CAUSES_FAILURE.
         nonempty_ground_nsrts = [nsrt for nsrt in ground_nsrts
                                  if nsrt.add_effects | nsrt.delete_effects]
-        if check_dr_reachable and \
-           not utils.is_dr_reachable(nonempty_ground_nsrts, atoms, task.goal):
+        all_reachable_atoms = utils.get_reachable_atoms(nonempty_ground_nsrts,
+                                                        atoms)
+        if check_dr_reachable and not task.goal.issubset(all_reachable_atoms):
             raise ApproachFailure(f"Goal {task.goal} not dr-reachable")
+        reachable_nsrts = [nsrt for nsrt in nonempty_ground_nsrts if
```

Comment:
This is an interesting change. I wonder if it has any substantial effect on oracle planning? Is the answer no because we typically don't have any unreachable NSRTs in oracle_approach.py?

Reply:
I didn't see any substantial effect. that's a good possible explanation. what do you think, worth keeping?